test/parallel/test-domain-intercept.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const domain = require('domain');
+
+{
+  const d = new domain.Domain();
+
+  const mustNotCall = common.mustNotCall();
+
+  d.on('error', common.mustCall((err) => {
+    assert.strictEqual(err.message, 'foobar');
+    assert.strictEqual(err.domain, d);
+    assert.strictEqual(err.domainEmitter, undefined);
+    assert.strictEqual(err.domainBound, mustNotCall);
+    assert.strictEqual(err.domainThrown, false);
+  }));
+
+  const bound = d.intercept(mustNotCall);
+  bound(new Error('foobar'));
+}
+
+{
+  const d = new domain.Domain();
+
+  const bound = d.intercept(common.mustCall((data) => {
```

Comment:
Some of these comments describe the intention of these tests. For instance: 

> // catch thrown errors no matter how many times we enter the event loop [...]

Looking at the new version of that test in `test-domain-implicit-binding.js`, it's not clear why the error is thrown from a nested async operation nested several layers deep, and so it's not clear to the person who hasn't written this test whether it even makes sense to keep this test.

I still think it would be worth it to keep those comments.